iroh-bytes/src/store/file.rs

```diff
@@ -0,0 +1,2483 @@
+            }
+            ActorMessage::Tags { filter, tx } => {
+                let txn = db.begin_read()?;
+                tx.send(self.tags(&ReadOnlyTables::new(&txn)?, filter)?)
+                    .ok();
+            }
+            ActorMessage::Get { hash, tx } => {
+                let txn = db.begin_read()?;
+                tx.send(self.get(&ReadOnlyTables::new(&txn)?, hash)).ok();
+            }
+            ActorMessage::EntryStatus { hash, tx } => {
+                let txn = db.begin_read()?;
+                tx.send(self.entry_status(&ReadOnlyTables::new(&txn)?, hash))
+                    .ok();
+            }
+            #[cfg(test)]
+            ActorMessage::EntryState { hash, tx } => {
+                let txn = db.begin_read()?;
+                tx.send(self.entry_state(&ReadOnlyTables::new(&txn)?, hash))
+                    .ok();
+            }
+            ActorMessage::Dump => {
+                let txn = db.begin_read()?;
+                dump(&ReadOnlyTables::new(&txn)?).ok();
+            }
+            ActorMessage::Validate {
+                progress,
+                repair,
+                tx,
+            } => {
+                let txn = db.begin_write()?;
+                let res = self.validate(&mut Tables::new(&txn)?, repair, progress);
+                txn.commit()?;
+                tx.send(res).ok();
+            }
+            ActorMessage::Import { cmd, tx } => {
+                let txn = db.begin_write()?;
+                let res = self.import(&mut Tables::new(&txn)?, cmd);
+                txn.commit()?;
+                tx.send(res).ok();
+            }
+            ActorMessage::Export { cmd, tx } => {
+                let txn = db.begin_write()?;
+                self.export(&mut Tables::new(&txn)?, cmd, tx)?;
+                txn.commit()?;
+            }
+            ActorMessage::CreateTag { hash, tx } => {
+                let txn = db.begin_write()?;
+                let res = self.create_tag(&mut Tables::new(&txn)?, hash);
+                txn.commit()?;
+                tx.send(res).ok();
+            }
+            ActorMessage::SetTag { tag, value, tx } => {
+                let txn = db.begin_write()?;
+                let res = self.set_tag(&mut Tables::new(&txn)?, tag, value);
+                txn.commit()?;
+                tx.send(res).ok();
+            }
+            ActorMessage::OnMemSizeExceeded { hash } => {
+                let txn = db.begin_write()?;
+                self.on_mem_size_exceeded(&mut Tables::new(&txn)?, hash)?;
+                txn.commit()?;
+            }
+            ActorMessage::OnComplete { hash } => {
+                let txn = db.begin_write()?;
+                self.on_complete(&mut Tables::new(&txn)?, hash)?;
+                txn.commit()?;
+            }
+            ActorMessage::HandleDropped { hash } => {
+                self.handle_dropped(hash);
+            }
+            ActorMessage::Sync { tx } => {
+                tx.send(()).ok();
```

Comment:
nothing to be done at all here?

Reply:
Yeah, the main purpose of sync is to just allow code to ensure that all messages before sync have been processed, and that we are not in a transaction.

Since this is not yet the batch branch, we are doing one transaction at a time. So all that remains is to just notify when the sync has been processed.